src/users/amoc_users_sup.erl

```diff
@@ -93,24 +100,24 @@ decr_no_of_users(SupNum) when SupNum > 1 ->
             ok
     end.
 
--spec start_child(amoc:scenario(), amoc_scenario:user_id(), any()) -> ok.
-start_child(Scenario, Id, ScenarioState) ->
-    Sup = get_sup_for_user_id(Id),
-    amoc_users_worker_sup:start_child(Sup, Scenario, Id, ScenarioState).
-
 -spec stop_child(pid(), boolean()) -> ok.
 stop_child(Pid, Force) ->
-    amoc_users_worker_sup:stop_child(Pid, Force).
+    case ets:lookup(?TABLE, Pid) of
+        [Object] ->
+            Sup = gen_sup_from_userid(Object),
+            amoc_users_worker_sup:stop_children(Sup, [Pid], Force);
+        _ ->
+            ok
+    end.
 
 %% Group all children based on ID to their respective worker supervisor and cast a request with each
 %% group at once. This way we reduce the number of casts to each worker to always one, instead of
 %% depending on the number of users.
--spec start_children(amoc:scenario(), [amoc_scenario:user_id()], any()) -> ok.
-start_children(Scenario, UserIds, ScenarioState) ->
-    State = persistent_term:get(?MODULE),
-    #storage{sups = Supervisors, sups_indexed = IndexedSups, sups_count = SupCount} = State,
-    Acc = maps:from_list([ {Sup, []} || {Sup, _} <- IndexedSups ]),
-    Assignments = assign_users_to_sups(SupCount, Supervisors, UserIds, Acc),
+-spec start_children(amoc:scenario(), {amoc_scenario:user_id(), amoc_scenario:user_id()}, any()) ->
+    ok.
+start_children(Scenario, {StartId, EndId}, ScenarioState) ->
```

Comment:
let's convert `StartId`, `EndId` into normal function parameters, tuple is unneeded here.